pkg/gitprovider/azure-devops.go

```diff
@@ -0,0 +1,472 @@
+	repo.Id = g.getAzureDevopsRepoId(repo.Source, repo.Owner, repo.Name, project)
+
+	return repo, nil
+}
+
+func (g *AzureDevOpsGitProvider) parseAzureDevopsHttpGitUrl(gitURL string) (*StaticGitContext, error) {
+	u, err := url.Parse(gitURL)
+	if err != nil {
+		return nil, err
+	}
+
+	repo := &StaticGitContext{}
+	urlPattern := `^(https?://)?(?P<source>[^/]+)/(?P<org>[^/]+)(?:/(?P<project>[^/_]+))?/_git/(?P<repo>[^/?]+)(?:\?.*)?(?:/.*)?$`
+	urlPatternRegex := regexp.MustCompile(urlPattern)
+	matches := urlPatternRegex.FindStringSubmatch(gitURL)
+	if len(matches) < 6 {
+		return nil, errors.New("cannot parse git URL: " + gitURL)
+	}
+
+	repo.Source = u.Host
+	repo.Owner = matches[3]
+	repo.Name = matches[5]
+	projectName := matches[4]
+
+	urlPath := strings.TrimPrefix(u.Path, "/")
+
+	if projectName == "" {
+		projectName = repo.Name
+		parts := strings.SplitN(urlPath, fmt.Sprintf("%s/", repo.Owner), 2)
+		if len(parts) != 2 {
+			return nil, errors.New("cannot parse git URL: " + gitURL)
+		}
+
+		urlPath = strings.Join([]string{parts[0], repo.Owner, projectName, parts[1]}, "/")
+		urlPath = strings.TrimPrefix(urlPath, "/")
+	}
+
+	parts := strings.Split(urlPath, "/")
+
+	repo.Url = g.getAzureDevopsCloneUrl(repo.Source, repo.Owner, repo.Name, projectName)
+	repo.Id = g.getAzureDevopsRepoId(repo.Source, repo.Owner, repo.Name, projectName)
+
+	queryParams, err := url.QueryUnescape(u.RawQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(parts) <= 4 && queryParams == "" {
+		return repo, nil
+	}
+
+	switch {
+	case len(parts) >= 6 && parts[4] == "pullrequest":
+		prNumber, _ := strconv.Atoi(parts[5])
+		prUint := uint32(prNumber)
+		repo.PrNumber = &prUint
+		repo.Path = nil
+	case len(parts) >= 6 && parts[4] == "commit":
+		repo.Sha = &parts[5]
+		repo.Path = nil
+	}
+
+	switch {
+	case strings.Contains(queryParams, "itemVersion=GB"):
+		fallthrough
+	case strings.Contains(queryParams, "version=GB"):
+		fallthrough
+	case strings.Contains(queryParams, "version=GT"):
+		fallthrough
+	case strings.Contains(queryParams, "refName=refs/heads/"):
+		re := regexp.MustCompile(`(itemVersion|version|refName)=(GB|GT|refs/heads/)(.+?)(&|$)`)
+		matches := re.FindStringSubmatch(queryParams)
+		if len(matches) != 5 {
+			return nil, errors.New("cannot parse git URL: " + gitURL)
+		}
+
+		repo.Branch = &matches[3]
+		repo.Path = nil
+	case strings.Contains(queryParams, "path="):
+		re := regexp.MustCompile(`path=/(.+?)(&|$)`)
+		matches := re.FindStringSubmatch(queryParams)
+		if len(matches) != 3 {
+			return nil, errors.New("cannot parse git URL: " + gitURL)
+		}
+
+		repo.Path = &matches[1]
+	}
+
+	return repo, nil
+}
+
+func (g *AzureDevOpsGitProvider) getAzureDevopsCloneUrl(source string, owner string, repo string, project string) string {
+	return fmt.Sprintf("https://%s/%s/%s/_git/%s", source, owner, project, repo)
+}
+
+func (g *AzureDevOpsGitProvider) getAzureDevopsRepoId(source string, owner string, repo string, project string) string {
```

Comment:
This function should return an error and have the "caller" function handle it (instead of `log.Fatal`)
I am still having issues `daytona create`-ing a repo URL:

![image](https://github.com/daytonaio/daytona/assets/25279767/857ede9a-bc36-446f-865f-57d43064f0c0)

The server shows this error and then the server process exits - this shouldn't happen.

![image](https://github.com/daytonaio/daytona/assets/25279767/17734a3c-3d02-4330-badf-be0f1e036d47)

Have you tried testing the git context parsing functions with one of your repos?